<launch>
  <arg name="play_bag" default="true" />
  <arg name="bag_file" default="/Users/helen/data/cow/small.bag"/>
  <arg name="voxel_size" default="0.02"/>
  <arg name="generate_esdf" default="false" />

  <node name="player" pkg="rosbag" type="play" output="screen" args=" -r 0.25 --clock $(arg bag_file)" if="$(arg play_bag)"/>

   <node name="voxblox_node" pkg="voxblox_ros" type="voxblox_node" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="points_processed"/>
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="color_mode" value="lambert" />
    <param name="use_tf_transforms" value="false" />
    <param name="update_mesh_every_n_sec" value="1.0" />
    <param name="min_time_between_msgs_sec" value="0.2" />
    <param name="method" value="merged" />
    <param name="anti_grazing" value="false">
    <param name="generate_esdf" value="$(arg generate_esdf)" />
    <param name="use_const_weight" value="false" />
    <param name="allow_clear" value="false" />
    <!--<param name="truncation_distance" value="0.5" /> -->
    <param name="verbose" value="true" />
    <remap from="transform" to="vrpn_client_kinect/raw_transform" />
    <rosparam file="$(find voxblox_ros)/cfg/cow_dataset.yaml"/>
    <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/$(anon cow).ply" />
  </node>

  <!-- <node name="octomap_manager" pkg="octomap_world"
      type="octomap_manager" output="screen" clear_params="true"
      args="-alsologtostderr" >
    <param name="resolution" value="0.1" />
    <param name="map_publish_frequency" value="5.0" />
    <param name="visualize_min_z" value="-1.0" />
    <param name="visualize_max_z" value="1.5" />
    <remap from="pointcloud" to="points_processed" />
  </node> -->
</launch>
